#35 add from to team request

ivshapovalov opened the pull request:
Connect #35

ivshapovalov committed:
#35 - split model classes to users and teams packages

ivshapovalov committed:
#35
- add from field to slack request
- change appropriate code

ivshapovalov committed:
#35
- update test classes for use 'from' field

ivshapovalov committed:
#35 - some refactoring

ivshapovalov committed:
#35
- test refactoring

adaleorion commented on src/main/java/ua/com/juja/microservices/teams/slackbot/service/impl/TeamServiceImpl.java:
```diff
@@ -41,27 +43,45 @@ public TeamServiceImpl(TeamRepository teamRepository, UserService userService) {
-        TeamRequest teamRequest = new TeamRequest(uuids);
-        Team activatedTeam = teamRepository.activateTeam(teamRequest);
-        checkTeamMembersEquality(teamRequest.getMembers(), activatedTeam.getMembers());
+        fromUser = SlackNameHandler.addAtToSlackName(fromUser);
```
Could you think about this code between lines 54 - 64?
Is this code simple for understanding and reading?
Could you change it for simple reading and understanding?
Maybe refactoring and some method extraction help you.
Could you try to do it?

adaleorion reviewed (changes requested):
nice work,
try to change some part of a code for simplifying reading and understanding

ivshapovalov committed:
#35 - simplify Service methods activate and deactivate Team

OlgaKulikova commented on src/main/java/ua/com/juja/microservices/teams/slackbot/model/teams/Team.java:
```diff
@@ -29,16 +34,27 @@
+        this.members = members;
+    }
 
-    @JsonCreator
```
Are you sure we don't need @JsonCreator annotation for any constructor?

OlgaKulikova commented on src/main/java/ua/com/juja/microservices/teams/slackbot/service/impl/TeamServiceImpl.java:
```diff
@@ -41,27 +41,51 @@ public TeamServiceImpl(TeamRepository teamRepository, UserService userService) {
+        log.debug("Before extract members from users : '{}'. Expected size '{}'. FromUser is '{}'", users,
+                expectedSize, fromUser);
+        Set<String> uuids;
+        if (users.size() == expectedSize) {
```
I'd better add a comments for these if-else to understand business logic. E.g. `if .... //if a fromUser is a part of the team` and `else .... //if a fromUser is not a part of the team`.
Because I spent some time to understand why you wrote such way.

ivshapovalov committed:
#35
- @JsonCreator on all constructors
- add comments to getMembers() method in TeamServiceImpl

ivshapovalov committed:
#35
- remove extra Team constructors
- remove extra annotations

ivshapovalov committed:
#35
- remove extra Team constructor. Remain only one @AllArgsConstructor
